### PR TITLE
EWPP-4515: Add new fields for Gallery pattern.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: templates/patterns/gallery/pattern-gallery.html.twig [hunk 1]
```diff
@@ -17,6 +17,11 @@
   } %}
 {% endif %}
 
+{% if grid %}
+  {% set columns = columns|default(2) %}
+  {% set ratio = ratio|default('3-2') %}
```
Comment (22Alexandra) [inline templates/patterns/gallery/pattern-gallery.html.twig]: If the same name is used for the variables in the template, we rely on the ECL implementation so checking if the `grid` is enabled wouldn't be needed. Instead let's use different variable names to be passed to the ECL component.
```suggestion
  {% set _columns = columns|default(2) %}
  {% set _ratio = ratio|default('3-2') %}
```

File: templates/patterns/gallery/pattern-gallery.html.twig [hunk 2]
```diff
@@ -63,6 +68,10 @@
   'expandable': expandable ?? true,
   'visible_items': visible_items|default(8),
   'footer': footer_link|default([]),
+  'grid': grid|default(false),
+  'column': columns|default(null),
+  'ratio': ratio|default(null),
```
Comment (22Alexandra) [inline templates/patterns/gallery/pattern-gallery.html.twig]: Considering the comment above, let's rename the passed variables for columns and ratio.
```suggestion
  'column': _columns|default(null),
  'ratio': _ratio|default(null),
```

File: templates/patterns/gallery/gallery.ui_patterns.yml [hunk 3]
```diff
@@ -146,3 +146,27 @@ gallery:
...
+      escape: false
+    columns:
+      type: "numeric"
+      label: "Columns number"
+      description: "Number of columns. Default is 2. Only used if the grid display is enabled."
+      preview: 2
+      escape: false
+    ratio:
+      type: "string"
+      label: "Column number"
```
Comment (22Alexandra) [inline templates/patterns/gallery/gallery.ui_patterns.yml]: It's not the column number.
```suggestion
      label: "Aspect ratio"
```

File: templates/patterns/gallery/gallery.ui_patterns.yml [hunk 4]
```diff
@@ -146,3 +146,27 @@ gallery:
...
+      description: "Image aspect ratio. Default is 3-2. Only used if the grid display is enabled."
+      preview: 2
```
Comment (22Alexandra) [inline templates/patterns/gallery/gallery.ui_patterns.yml]: This value for the preview is wrong and it's not working.

File: templates/patterns/gallery/gallery.ui_patterns.yml [hunk 5]
```diff
@@ -146,3 +146,27 @@ gallery:
...
+      escape: false
```
Comment (22Alexandra) [inline templates/patterns/gallery/gallery.ui_patterns.yml]: Do we need to escape a string?
Let's also add the empty new line at the end of the file.

File: tests/src/Kernel/fixtures/rendering.yml [hunk 6]
```diff
@@ -785,6 +786,14 @@
       '.ecl-gallery .ecl-button.ecl-button--ghost.ecl-gallery__slider-previous': 0
       '.ecl-gallery .ecl-button.ecl-button--ghost.ecl-gallery__slider-next': 0
       '.ecl-gallery__footer-link': 0
+      '.ecl-gallery--grid': 0
+      '.ecl-gallery--col-2': 0
+      '.ecl-gallery--col-3': 0
+      '.ecl-gallery--col-4': 0
+      '.ecl-gallery--col-5': 0
+      '.ecl-gallery--ratio-3-2': 0
+      '.ecl-gallery--ratio-3-1': 0
```
Comment (22Alexandra) [inline tests/src/Kernel/fixtures/rendering.yml:816]: Values for `columns` and `ratio` should be provided in order to assert they are not used if grid is not enabled.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]
Let's fix the ticket number in the commit message and PR title.

**Review by 22Alexandra — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

**Pull request merged**